### PR TITLE
Publish custom integrations and add-ons

### PR DESCRIPTION
**Pull request by ludeeus**

### CODE REVIEW

File: worker/src/data.ts [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
 export const KV_KEY_CORE_ANALYTICS = "core_analytics";
+export const KV_KEY_CUSTOM_INTEGRATIONS = "custom_integrations";
+export const KV_KEY_ADDONS = "addons";
 export const KV_PREFIX_HISTORY = "history";
 export const KV_PREFIX_UUID = "uuid";
 export const KV_MAX_PROCESS_ENTRIES = 850;
 
-export const SCHEMA_VERSION_QUEUE = 1;
+export const BRANDS_DOMAINS_URL =
+  "http://brands.home-assistant.io/domains.json";
```
Comment (balloob) [inline worker/src/data.ts:10]:
```suggestion
  "https://brands.home-assistant.io/domains.json";
```

Comment (balloob) [reply worker/src/data.ts:10]: Also https://github.com/home-assistant/brands/pull/2536#issuecomment-832417273

File: worker/src/handlers/schedule.ts [hunk 2]
```diff
@@ -328,6 +348,59 @@ function combineEntryData(
...
+        data.addons[addon.slug].versions[addon.version] = bumpValue(
+          data.addons[addon.slug].versions[addon.version]
+        );
+      }
+    }
+  }
+
+  if (reported_custom_integrations.length) {
+    for (const custom_integration of reported_custom_integrations) {
+      if (!brandsDomains.includes(custom_integration.domain)) {
```
Comment (balloob) [inline worker/src/handlers/schedule.ts:381]: Let's also add this check to normal integrations.

**Review by balloob — APPROVED** [review]

File: worker/src/handlers/schedule.ts [hunk 3]
```diff
@@ -180,12 +183,20 @@ async function processQueue(sentry: Toucan): Promise<void> {
     }
   }
 
+  sentry.addBreadcrumb({ message: "Fetching domain list from brands" });
+  const brandsDomainsResponse = await fetch(BRANDS_DOMAINS_URL);
+  if (!brandsDomainsResponse.ok) {
+    throw Error("Could not get domain list from brands");
+  }
+
+  const brandsDomains: string[] = await brandsDomainsResponse.json();
```
Comment (balloob) [inline worker/src/handlers/schedule.ts:192]: Format has changed. 

Also, faster to make this a Set.

**Review by balloob — APPROVED** [review]

**Comment by ludeeus** [comment]
Until secrets are added, and the files are generated the first time, it's expected that CI/build scripts fails, this PR will never pass CI

**Review by balloob — APPROVED** [review]

File: worker/src/handlers/schedule.ts [hunk 4]
```diff
@@ -223,6 +241,12 @@ async function processQueue(sentry: Toucan): Promise<void> {
       JSON.stringify(queue_data)
     );
     await KV.put(KV_KEY_CORE_ANALYTICS, JSON.stringify(storedAnalytics));
+    await KV.put(
```
Comment (balloob) [inline worker/src/handlers/schedule.ts:244]: Run all these `puts` in parallel. `await Promise.all(...)`

**Pull request merged**